### PR TITLE
feat: refactoring and add copyrightZoneTitle attribute for customization

### PR DESCRIPTION
**Pull request by shc261392**
## Added
- `cz-zone` attribute to override default copyright zone title

## Chore
- Refactor `modal-manager.ts` and `modal.ts`, and add more tests

### CODE REVIEW

File: src/modal/modal.ts [hunk 1]
```diff
@@ -48,24 +59,35 @@ export class CaptureEyeModal extends LitElement {
...
-      this._imageLoaded = false;
-      this.engagementImage = engagementImage;
-      this.requestUpdate();
+  updateModalOptions(options: ModalOptions) {
+    if (options.nid) this.nid = options.nid;
+    if (options.layout) this.layout = options.layout;
+    if (options.copyrightZoneTitle)
+      this._copyrightZoneTitle = options.copyrightZoneTitle;
+    if (options.engagementImage)
+      this._engagementImage = options.engagementImage;
```
Comment (olgahaha) [inline src/modal/modal.ts:68]: Just to confirm, does _imageLoaded need to be cleared? The previous method reset the value when the engagementImage changed, but now there's no place where the value is being reset.

File: src/modal/modal.ts [hunk 2]
```diff
@@ -366,8 +393,7 @@ export class CaptureEyeModal extends LitElement {
     // 0: User-customized link (not provided by us)
     // 1: Default engagement link
     // 2, 3, ...: Future rotating engagement links
-    const subid =
-      this.engagementLink === Constant.url.defaultEngagementLink ? '1' : '0';
+    const subid = this._engagementLink ? '0' : '1';
```
Comment (olgahaha) [inline src/modal/modal.ts:396]: The default value is defaultEngagementLink, so it's probably best to keep the original logic. Also, when the user customizes the value to match our default link, the value should be set to '1'.

Comment (shc261392) [reply src/modal/modal.ts:396]: @olgahaha In this update I changed the protected states like `this._engagementLink` to override-only (defaults to empty string. If it's empty then when rendering use default value. Only if not empty then use the custom value for overriding. So the new default value of `this._engagementLink` is empty, but I can see your point we should also treat when `eng-link` attribute is set to the same value as default link, we should also set it to 1.

File: src/modal/modal.ts (hunk 1 above)
Comment (shc261392) [reply src/modal/modal.ts:68]: Yes I believe it should be cleared as well. Let me check and update it.

Comment (olgahaha) [reply src/modal/modal.ts:68]: I couldn’t find where it was cleared. Could you point it out for me? Thanks!

Comment (shc261392) [reply src/modal/modal.ts]: @olgahaha Updated in the latest commit

**Review by olgahaha — APPROVED** [review]

**Pull request merged**